src/entities/trades/swapr-v3/SwaprV3.ts

```diff
@@ -101,26 +107,18 @@ export class SwaprV3Trade extends TradeWithSwapTransaction {
           quoteCurrency.name,
         )
 
-    invariant(
-      (await provider.getNetwork()).chainId == chainId,
-      `SwaprV3Trade.getQuote: currencies chainId does not match provider's chainId`,
-    )
-
-    const routes = await getRoutes(tokenIn, tokenOut, chainId)
+    const routes = await getRoutes(setToken, quoteToken, chainId)
 
     const fee =
       routes?.length > 0 && routes[0].pools.length > 0
         ? new Percent(routes[0].pools[0].fee.toString(), '1000000')
```

Comment:
Let's also use `ALGEBRA_FEE_PARTS_PER_MILLION` here, instead of `'1000000'`.

Reply:
const ALGEBRA_FEE_PARTS_PER_MILLION = JSBI.BigInt(1_000_000)

this is different then "1000000"